packages/flutter_tools/lib/src/ios/cocoapods.dart

```diff
@@ -60,13 +61,10 @@ class CocoaPods {
     // For backward compatibility with previously created Podfile only.
     @required String iosEngineDir,
     bool isSwift: false,
-    bool pluginOrFlutterPodChanged: true,
+    bool flutterPodChanged: true,
   }) async {
     if (await _checkPodCondition()) {
-      if (!fs.file(fs.path.join(appIosDir.path, 'Podfile')).existsSync()) {
```

Comment:
I'm assuming this implies we've built in some knowledge here that `flutter run` will necessarily run `flutter packages get` before and will never have cases where `pod install` is run in a project that doesn't have a Podfile?

Reply:
That's the idea. `flutter run` will automatically call `flutter packages get` if your `pubspec.yaml` file has changed relative to your `.packages` and/or `.pubspec.lock`. So you'll get a `Podfile` if your project needs it and you don't already have one; it just isn't done by our `buildXcodeProject` code path.

I guess I should add checks for those assumptions here.

Reply:
A check would be good to proof against future refactors by maintainers unaware of this contract.